CppImport/src/macro/NodeMapping.h

```diff
@@ -0,0 +1,88 @@
+/***********************************************************************************************************************
+ **
+ ** Copyright (c) 2011, 2015 ETH Zurich
+ ** All rights reserved.
+ **
+ ** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+ ** following conditions are met:
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#pragma once
+
+#include "cppimport_api.h"
+
+#include <ModelBase/src/nodes/Node.h>
+
+namespace CppImport {
+
+class CPPIMPORT_API NodeMapping
+{
+	public:
+		void add(Model::Node* original, Model::Node* clone)
+		{
+			Q_ASSERT(original);
+			Q_ASSERT(clone);
+			clones_[clone] = original;
+			originals_[original] = clone;
+		}
+
+		Model::Node* original(Model::Node* clone)
+		{
+			if (!clones_.contains(clone)) return nullptr;
```

Comment:
Again, use `find` instead of `contains`+`value`, to avoid redundancy.

Please do a full text search for `contains` in your code and examine each use to see if you can replace it with `find` and eliminate redundant searches.